judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/AbstractTest.java

```diff
@@ -315,76 +313,71 @@ void testAbstractInheritedEntityRelationsAndCompositionsOnANotAbstractEntity() {
         iMultiRelation.forEach(i -> i.setNameI("iMultiRelationNameChanged"));
 
-
-        iDao.update(h.getCompositionIOnHSingle().get());
-        iDao.update(h.getCompositionIOnHSingleRequired());
-        h.setCompositionIOnHMulti(h.getCompositionIOnHMulti().stream().map(i -> iDao.update(i)).collect(Collectors.toList()));
-
         iOptionalRelation = iDao.update(iOptionalRelation);
         iRequiredRelation = iDao.update(iRequiredRelation);
         iMultiRelation = iMultiRelation.stream().map(i -> iDao.update(i)).collect(Collectors.toList());
 
         h = hDao.getById(h.identifier()).orElseThrow();
 
         //Check update
-        assertEquals(Optional.of("CompositionSingleChanged"), h.getCompositionIOnHSingle().orElseThrow().getNameI());
-        assertEquals(Optional.of("CompositionSingleRequiredChanged"), h.getCompositionIOnHSingleRequired().getNameI());
-        h.getCompositionIOnHMulti().stream().forEach(i -> assertEquals(Optional.of("MultiCompositionNameChanged"), i.getNameI()));
-        hDao.queryCompositionIOnHMulti(h).execute().stream().forEach(i -> assertEquals(Optional.of("MultiCompositionNameChanged"), i.getNameI()));
 
         assertEquals(Optional.of("iOptionalRelationNameChanged"), hDao.queryRelationIOnHSingle(h).orElseThrow().getNameI());
         assertEquals(Optional.of("iRequiredRelationNameChanged"), hDao.queryRelationIOnHSingleRequired(h).getNameI());
         hDao.queryRelationIOnHMulti(h).execute().stream().forEach(i -> assertEquals(Optional.of("iMultiRelationNameChanged"), i.getNameI()));
 
-        //delete with IDao
+        //delete relations with IDao
 
-        iDao.delete(h.getCompositionIOnHSingle().get());
-        assertTrue(iDao.getById(h.getCompositionIOnHSingle().get().identifier()).isEmpty());
-        assertTrue(jDao.getById(h.getCompositionIOnHSingle().get().identifier().adaptTo(JIdentifier.class)).isEmpty());
+        I deleteSingleRelation = hDao.queryRelationIOnHSingle(h).orElseThrow();
+        iDao.delete(deleteSingleRelation);
+        assertTrue(iDao.getById(deleteSingleRelation.identifier()).isEmpty());
+        assertTrue(jDao.getById(deleteSingleRelation.identifier().adaptTo(JIdentifier.class)).isEmpty());
         assertEquals(iDao.getAll().size(), jDao.getAll().size());
 
-        //TODO JNG-4812
-//        iDao.delete(h.getCompositionIOnHSingleRequired()); //unfortunately you can delete required Component
-//        h = hDao.getById(h.identifier()).orElseThrow();
 
-        iDao.delete(h.getCompositionIOnHMulti().get(0));
-        assertTrue(iDao.getById(h.getCompositionIOnHMulti().get(0).identifier()).isEmpty());
+        I deleteSingleRequiredRelation = hDao.queryRelationIOnHSingleRequired(h);
+        IllegalStateException thrown = assertThrows(
+                IllegalStateException.class,
+                () -> iDao.delete(deleteSingleRequiredRelation)
+        );
+
+        assertTrue(thrown.getMessage().contains("There is mandatory references which is not removable"));
```

Comment:
"There are", "which are not"

Reply:
You need to change the thrown message if you want this changes